src/steps/change_directory_step.go

```diff
@@ -27,7 +28,9 @@ func (step *ChangeDirectoryStep) Run() error {
 	_, err := os.Stat(step.Directory)
 	if err == nil {
 		script.PrintCommand("cd", step.Directory)
-		return os.Chdir(step.Directory)
+		if !dryrun.IsActive() {
```

Comment:
I prefer clean guard clauses for dry-run mode rather than hacking it into the business logic of steps.

Reply:
I'll remove this one. I think the one in the push step is necessary as otherwise dry run doesn't include pushes which I think are good to have